UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/WebhookController.cs

```diff
@@ -30,12 +43,49 @@ public IActionResult NewFilesPublishedOptions()
         [Route("/webhook/newfilespublished")]
         public virtual async Task<IActionResult> NewFilesPublished()
         {
-            ////Adding payload to log temporarily so that in can be used for unit testing etc. 
-            using var reader = new StreamReader(Request.Body, Encoding.UTF8);
+            using StreamReader reader = new(Request.Body, Encoding.UTF8);
             string payload = await reader.ReadToEndAsync();
-            Logger.LogInformation(EventIds.ClearFSSSearchCacheEventStart.ToEventId(), "Clear FSS Search Cache Event started for _X-Correlation-ID:{correlationId} Payload: {payload}", GetCurrentCorrelationId(), payload);
 
-            return GetCacheResponse(); 
+            _logger.LogInformation(EventIds.ClearFSSSearchCacheEventStarted.ToEventId(), "Clear FSS search cache event started for _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+            if (string.IsNullOrEmpty(payload))
+            {
+                _logger.LogInformation(EventIds.ClearFSSSearchCacheValidationEvent.ToEventId(), "Payload is null or empty for Enterprise event _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+                return GetCacheResponse();
+            }
+            EventGridEvent eventGridEvent = new();
+
+            JsonConvert.PopulateObject(payload, eventGridEvent);
+            if (eventGridEvent.Data == null || eventGridEvent.Data.ToString() == "{}")
```

Comment:
Kept null check for eventGridEvent.Data as it contains required information like business unit and product type required to clear cache

Reply:
Thanks for the update, @Avinash15515.

I agree that the null check is important here, as is the check for an empty JSON payload.

However, there is no check in the same statement for empty strings; an empty byte array from within BinaryData would return this, we should guard against this _as well_.

Reply:
```suggestion
            if (string.IsNullOrEmpty(eventGridEvent.Data?.ToString()) || eventGridEvent.Data.ToString() == "{}")
```

Here is an example of what I mean about extra blank-string check. In a longer form, this would be:

`if (eventGridEvent.Data == null || eventGridEvent.Data.ToString() == "" || eventGridEvent.Data.ToString() == "{}")`

Hope this helps

Reply:
Thanks, That certainly helped. Have added the check. Also added the unit test for same.